channels/pjsip/dialplan_functions.c

```diff
@@ -204,6 +204,25 @@
 		<para>Parse an URI and return a specified part of the URI.</para>
 	</description>
 </function>
+<function name="PJSIP_HANGUP" language="en_US">
+	<synopsis>
+		Hangup an incoming PJSIP channel with a SIP response code
+	</synopsis>
+	<syntax>
+		<parameter name="response_code" required="true">
+			<para>SIP response code between 400 and 699.</para>
+		</parameter>
+	</syntax>
+	<description>
+		<para>Hangs up an incoming PJSIP channel and returns the
+		specified SIP response code to the caller.</para>
+		<warning><para>
+		This function must be called BEFORE anything that
+		might cause any other final (non 1XX) response to be sent.
+		</para></warning>
```

Comment:
An example would be nice.

e.g. `exten => s,1,Set(PJSIP_HANGUP()=404)`

And I think people might want to know if they can do other stuff after calling this, will dialplan execution end immediately after?

Are there examples of things that might cause a final response to be sent?